From 7d9fadfa222731d5e0c0bcd2320b01eea56465b3 Mon Sep 17 00:00:00 2001
From: Robert Yang <liezhi.yang@windriver.com>
Date: Mon, 5 May 2014 23:24:14 +0800
Subject: [PATCH] anaconda: run sync() before shutdown or reboot

We need flush file system buffers before shutdown or reboot, otherwise
the data may incomplete.

Upstream-Status: Pending

Signed-off-by: Robert Yang <liezhi.yang@windriver.com>
---
 anaconda | 1 +
 1 file changed, 1 insertion(+)

diff --git a/anaconda b/anaconda
index 25e2086..53e421e 100755
--- a/anaconda
+++ b/anaconda
@@ -782,6 +782,7 @@ if __name__ == "__main__":
             while True:
                 time.sleep(10000)
 
+    os.system("sync")
     if anaconda.ksdata:
         from pykickstart.constants import KS_SHUTDOWN, KS_WAIT, KS_REBOOT
 
-- 
1.8.3.1

